### PR TITLE
dev: add internal dev section to release notes

### PR DESCRIPTION
**Pull request by Kuchenpirat**
## What type of PR is this?

- dev

## What this PR does / why we need it:

This pull request introduces a new section in our release notes dedicated to internal development PRs. This change will help separate pull requests that don't alter Mealie itself.

I reordered the PR types in the Pull request template to match the PR title section which seemed more logical to me. 
Once this is merged, I will update the titles of unreleased PRs to ensure they are categorized correctly.

## Which issue(s) this PR fixes:

None

### CODE REVIEW

File: .github/release-drafter.yml [hunk 1]
```diff
@@ -29,6 +29,9 @@ categories:
     collapse-after: 3
     labels:
       - "dependencies"
+  - title: "🔨 Internal development"
+    labels:
+      - "dev"
```
Comment (boc-the-git) [inline .github/release-drafter.yml]: Thoughts on putting this above the dependencies? I see this as something that at least _some_ people would want to look at, whereas no one cares about the dependencies

Comment (Kuchenpirat) [reply .github/release-drafter.yml]: sure 😅👍

File: .github/workflows/pull-request-lint.yml [hunk 2]
```diff
@@ -20,10 +20,11 @@ jobs:
...
           types: |
-            fix
-            feat
-            docs
-            chore
+            feature
+            bug
+            documentation
+            cleanup
+            dev
```
Comment (boc-the-git) [inline .github/workflows/pull-request-lint.yml]: This is checking the string prefix for the title, not labels. So `feat` for a `feat: xyz` title, not `feature` which is the label that gets added

Comment (boc-the-git) [reply .github/workflows/pull-request-lint.yml]: You only need to add `dev` to the existing list, and not change anything else, I believe

Comment (Kuchenpirat) [reply .github/workflows/pull-request-lint.yml]: yeah, i copied the wrong section

**Review by boc-the-git — APPROVED** [review]

**Comment by Kuchenpirat** [comment]
all PRs since the last release should now be labeled correctly

**Pull request merged**